### PR TITLE
Refactoring code to use bower/lib/config instead and fix wrong tests.

### PR DESCRIPTION
**Pull request by henvic**
Your current logic doesn't account for the merging strategy described in the [config docs](bower.io/docs/config/).

### CODE REVIEW

**Comment by shinnn** [comment]
Test failed.

**Comment by henvic** [comment]
Oops. I have just fixed it.

Can you run again the AppVeyor test for 0.10? It seems to be a temporary issue with their system. https://ci.appveyor.com/project/ShinnosukeWatanabe/bower-directory/build/25/job/lruasxu4803ytu5l

**Comment by henvic** [comment]
Everything is passing now.

File: index.js [hunk 1]
```diff
@@ -1,48 +1,32 @@
...
-  if (isFileSync(bowerrcPath)) {
-    return resolveFromCwd(getBowerDir(readJsonSync(bowerrcPath)));
-  } else {
-    return resolveFromCwd('bower_components');
+  try {
+    p = bowerDirectorySync(option);
+  } catch (e) {
+    err = e;
+  } finally {
+    cb(err, p);
```
Comment (shinnn) [inline index.js]: I want `bowerDirectory(option, cb)` to work as an asynchronous function, but this change makes it synchronous...

**Comment by henvic** [comment]
@shinnn, I have wrapped the call with a setTimeout on a new commit (d8583498155aa55b0d54231f13af11ae9eac066e) to make it non-blocking as you want.

**Comment by henvic** [comment]
Any news?

**Comment by shinnn** [comment]
Closing as the upstream project Bower is no longer maintained.

**Pull request closed without merging**